### PR TITLE
[tensorflow ] Upgrade to 2.9.1 and support osx-arm64.

### PR DESCRIPTION
**Pull request by matt-42**
This PR upgrade tensorflow to 2.9.1 and add support of triplet osx-arm64

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
existing triplets plus osx-arm64

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
 Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/tensorflow-cc/vcpkg.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "tensorflow-cc",
-  "version-semver": "2.7.0",
+  "version-semver": "2.9.1",
   "port-version": 2,
```
Comment (JonLiu1993) [inline ports/tensorflow-cc/vcpkg.json:4]:
```suggestion
```

File: ports/tensorflow-common/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "tensorflow-common",
-  "version-semver": "2.7.0",
+  "version-semver": "2.9.1",
   "port-version": 2,
```
Comment (JonLiu1993) [inline ports/tensorflow-common/vcpkg.json:4]:
```suggestion
```

File: ports/tensorflow/vcpkg.json [hunk 3]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "tensorflow",
-  "version-semver": "2.7.0",
+  "version-semver": "2.9.1",
   "port-version": 2,
```
Comment (JonLiu1993) [inline ports/tensorflow/vcpkg.json:4]:
```suggestion
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/tensorflow-common/tensorflow-common.cmake [hunk 4]
```diff
@@ -43,13 +43,34 @@ if(CMAKE_HOST_WIN32)
...
+		endif()
+	endif()
+
+	# create virtualenv.
+	message(STATUS "Create python virtualenv")
+	if (VCPKG_TARGET_IS_LINUX)
+				vcpkg_execute_required_process(COMMAND virtualenv --symlinks "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-venv"  WORKING_DIRECTORY ${CURRENT_BUILDTREES_DIR} LOGNAME prerequisites-venv-${TARGET_TRIPLET})
+	else()
+				vcpkg_execute_required_process(COMMAND ${PYTHON3} -m venv --symlinks "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-venv"  WORKING_DIRECTORY ${CURRENT_BUILDTREES_DIR} LOGNAME prerequisites-venv-${TARGET_TRIPLET})
+	endif()
```
Comment (JonLiu1993) [inline ports/tensorflow-common/tensorflow-common.cmake:63]: @matt-42 ,Please keep the code format here the same as above, the spaces here are too large

File: ports/tensorflow-cc/vcpkg.json [hunk 5]
```diff
@@ -1,10 +1,9 @@
 {
   "name": "tensorflow-cc",
-  "version-semver": "2.7.0",
-  "port-version": 2,
+  "version-semver": "2.9.1",
   "description": "Library for computation using data flow graphs for scalable machine learning (C++ API version)",
   "homepage": "https://github.com/tensorflow/tensorflow",
-  "supports": "!(x86 | arm | uwp)",
+  "supports": "!(x86 | arm | uwp) | (arm64 & osx)",
```
Comment (JonLiu1993) [inline ports/tensorflow-cc/vcpkg.json:6]:
```suggestion
  "supports": "!(x86 | arm | uwp) | (arm64 & osx)",
  "license": "MIT",
```

Comment (matt-42) [reply ports/tensorflow-cc/vcpkg.json:6]: Shouldn't it be "Apache-2.0" ? Tensorflow is under Apache-2.0.

Comment (JonLiu1993) [reply ports/tensorflow-cc/vcpkg.json:6]: @matt-42, I referenced [this](https://github.com/FloopCZ/tensorflow_cc/blob/master/LICENSE)

Comment (matt-42) [reply ports/tensorflow-cc/vcpkg.json:6]: Why referencing this ?  I do not see how the tensorflow-cc port is related to FloopCZ/tensorflow_cc

Comment (JonLiu1993) [reply ports/tensorflow-cc/vcpkg.json:6]: @matt-42 ,Sorry my mistake, it should be "Apache-2.0"

**Comment by JonLiu1993** [comment]
@matt-42, Ci tested failed in the x64-linux and x64-osx triplet,
This is error log, please take a look:
````
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:83 (message):
    Command failed: virtualenv --symlinks /mnt/vcpkg-ci/buildtrees/tensorflow/x64-linux-venv
    Working Directory: /mnt/vcpkg-ci/buildtrees/tensorflow
    Error code: No such file or directory
    See logs for more information:
````

**Comment by matt-42** [comment]
Hi @JonLiu1993 , I don't have more time to dedicate to this pr. I'm closing it. If anyone wants to take over feel free to start from my branch:  https://github.com/matt-42/vcpkg/tree/tensorflow-update-osx-arm64

**Pull request closed without merging**